govuk-notify-spi/src/main/scala/uk/gov/nationalarchives/notifyspi/NotifyEmailSenderProvider.scala

```diff
@@ -1,30 +1,43 @@
 package uk.gov.nationalarchives.notifyspi
 
 import java.util
-
 import org.keycloak.email.{EmailException, EmailSenderProvider}
 import org.keycloak.models.UserModel
+import software.amazon.awssdk.http.apache.ApacheHttpClient
+import software.amazon.awssdk.regions.Region
 import uk.gov.service.notify.NotificationClient
 
+import java.net.URI
 import scala.jdk.CollectionConverters._
 import scala.util.{Failure, Success, Try}
+import software.amazon.awssdk.services.ssm.SsmClient
+import software.amazon.awssdk.services.ssm.model.GetParameterRequest
+import com.typesafe.config.{ConfigFactory, Config => TypeSafeConfig}
 
 class NotifyEmailSenderProvider(environmentVariables: Map[String, String]) extends EmailSenderProvider {
+  val configFactory: TypeSafeConfig = ConfigFactory.load
+  val ssmEndpoint: String = configFactory.getString("ssm.endpoint")
+  val notifyApiKeyPath: String = configFactory.getString("notify.apiKeyPath")
+  val notifyTemplateIdPath: String = configFactory.getString("notify.templateIdPath")
 
   private implicit class EnvironmentalVariablesUtils(map: Map[String, String]) {
```

Comment:
You don't need this implicit class any more or the environmentVariables map passed into the constructor if you're using the typesafe config. So you can move the `getApiKey`, `getTemplateId` and `getSsmParameterValue` to the class `NotifyEmailSenderProvider` level and delete the implicit class.